src/NHibernate/Cache/Entry/StructuredMapCacheEntry.cs

```diff
@@ -8,7 +8,7 @@ public class StructuredMapCacheEntry : ICacheEntryStructure
 		public object Structure(object item)
 		{
 			CollectionCacheEntry entry = (CollectionCacheEntry)item;
-			object[] state = entry.State;
+			object[] state = (object[])entry.DisassembledState;
```

Comment:
> So, the `//TODO: assumes all collections disassemble to an array!` has moved here indeed.

And here too. Then why not keeping it in the `CollectionCacheEntry` class?

Reply:
You are right, I got confused as `IPersistentCollection.InitializeFromCache` and `IPersistentCollection.Disassemble` are using `object` which I think it should be changed to `object[]` in the future as all implementations are casting/returning it.